frame/contracts/src/lib.rs

```diff
@@ -344,6 +344,11 @@ pub trait Config: frame_system::Config {
 
 	/// The maximum amount of weight that can be consumed per block for lazy trie removal.
 	type DeletionWeightLimit: Get<Weight>;
+
+	/// The maximum length of a contract code in bytes. This limit applies to the instrumented
+	/// version of the code. Therefore `instantiate_with_code` can fail even when supplying
+	/// a wasm binary below this maximum size.
+	type MaxCodeSize: Get<u32>;
```

Comment:
when code is reinstumented, could it happen that it goes beyond max code size ?
reading from the code it seems we don't check for it. But also it shouldn't be too much of an issue.

Reply:
Yes this could happen but we couldn't throw any error in this case because the contract was already deployed successfully and this would break existing contracts. However, it will be instrumented from the same original code and the contents of the schedule do not influence the size of the instrumented code. Without changes to the instrumentation code this should not be an issue.